src/diffusers/models/attention.py

```diff
@@ -527,6 +527,62 @@ def forward(
         return hidden_states
 
 
+class LuminaFeedForward(nn.Module):
+    r"""
+    A feed-forward layer.
+
+    Parameters:
+        hidden_size (`int`):
+            The dimensionality of the hidden layers in the model. This parameter determines the width of the model's
+            hidden representations.
+        intermediate_size (`int`): The intermediate dimension of the feedforward layer.
+        multiple_of (`int`, *optional*): Value to ensure hidden dimension is a multiple
+            of this value.
+        ffn_dim_multiplier (float, *optional*): Custom multiplier for hidden
+            dimension. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        hidden_size: int,
+        intermediate_size: int,
+        multiple_of: Optional[int] = 256,
+        ffn_dim_multiplier: Optional[float] = None,
+    ):
+        super().__init__()
+        intermediate_size = int(2 * intermediate_size / 3)
+        # custom hidden_size factor multiplier
+        if ffn_dim_multiplier is not None:
+            intermediate_size = int(ffn_dim_multiplier * intermediate_size)
+        intermediate_size = multiple_of * ((intermediate_size + multiple_of - 1) // multiple_of)
+
+        self.w1 = nn.Linear(
+            hidden_size,
+            intermediate_size,
+            bias=False,
+        )
+        self.w2 = nn.Linear(
+            intermediate_size,
+            hidden_size,
+            bias=False,
+        )
+        self.w3 = nn.Linear(
+            hidden_size,
+            intermediate_size,
+            bias=False,
+        )
+        nn.init.xavier_uniform_(self.w1.weight)
+        nn.init.xavier_uniform_(self.w2.weight)
+        nn.init.xavier_uniform_(self.w3.weight)
```

Comment:
```suggestion
```

we don't need to consider initialization - the models and pipelines are for inference only. If you will share a training script later, the initialization can be handled there

Reply:
okay, done.